modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_5_1/test/DDMFieldUpgradeProcessTest.java

```diff
@@ -112,6 +118,89 @@ public void testUpgradeProcess() throws Exception {
 			ddmStructureVersion.getCompanyId(), ddmField.getCompanyId());
 	}
 
+	@Test
+	public void testUpgradeProcessSeveralCompanies() throws Exception {
+		Company company = CompanyTestUtil.addCompany();
+
+		try {
+			DDMForm ddmForm = DDMFormTestUtil.createDDMForm(
+				RandomTestUtil.randomString());
+
+			DDMStructure ddmStructure = _ddmStructureLocalService.addStructure(
+				TestPropsValues.getUserId(), _group.getGroupId(), 0,
+				PortalUtil.getClassNameId(DDLRecordSet.class.getName()),
+				"CUSTOM-META-TAGS", RandomTestUtil.randomLocaleStringMap(),
+				null, ddmForm, _ddm.getDefaultDDMFormLayout(ddmForm),
+				StorageType.DEFAULT.toString(),
+				DDMStructureConstants.TYPE_DEFAULT,
+				ServiceContextTestUtil.getServiceContext());
+
+			List<DDMStructureVersion> structureVersions =
+				_ddmStructureVersionLocalService.getStructureVersions(
+					ddmStructure.getStructureId());
+
+			DDMStructureVersion ddmStructureVersion = structureVersions.get(0);
+
+			DDMField ddmField = _addDDMField(
+				ddmStructureVersion.getStructureVersionId());
+
+			Assert.assertEquals(
+				CompanyConstants.SYSTEM, ddmField.getCompanyId());
+
+			_runUpgrade();
+
+			ddmField = _ddmFieldLocalService.getDDMField(ddmField.getFieldId());
+
+			Assert.assertEquals(
+				ddmStructureVersion.getCompanyId(), ddmField.getCompanyId());
+		}
+		finally {
+			_companyLocalService.deleteCompany(company);
+		}
+	}
+
+	@Test
+	public void testUpgradeProcessWithDatabasePartitionEnabled()
+		throws Exception {
+
+		Assume.assumeTrue(DBPartition.isPartitionEnabled());
+
+		DDMField ddmField = _addDDMField();
+
+		Assert.assertEquals(CompanyConstants.SYSTEM, ddmField.getCompanyId());
+
+		long companyId = RandomTestUtil.randomLong();
+
+		CompanyThreadLocal.setCompanyId(companyId);
```

Comment:
Hi @AliciaGarciaGarcia!

Thank you so much for the quick changes, they are going to be really useful for our customers.

Regarding the tests, I only want to mention that you do not need those two lines (probably the CompanyThreadLocal will throw an exception since checks the user of the company). You can just verify later that the companyId of the ddmField is the default one (we run the upgrade process once per company when db partition enabled).

Reply:
ok!